omod/src/main/java/org/openmrs/module/owa/filter/OwaFilter.java

```diff
@@ -49,23 +50,31 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
 			requestURL = request.getServletPath();
 		}
 		
+		String loginUrl = Context.getAdministrationService().getGlobalProperty("login.url", "login.htm");
 		if (Context.isAuthenticated()) {
 			if (requestURL.startsWith(owaBasePath)) {
 				String newURL = requestURL.replace(owaBasePath, "/ms/owa/fileServlet");
 				req.getRequestDispatcher(newURL).forward(req, res);
 			} else {
-				chain.doFilter(req, res);
+				if (loginUrl.contains("addonmanager") && (request.getServletPath().equals("/index.htm") || request.getServletPath().equals("/login.htm") || request.getServletPath().equals("/"))) {

```

Comment:
Isn't this a duplicate of what you have on line 73 and below?

Reply:
No, this care for if one re-invokes /index.htm, /login.htm or / when already authenticated to still take them back to the addonmanager landing page if the conditions are fulfilled.

Reply:
What i mean is that you are duplicating the logic in your if condition. Does this make sense?

Reply:
Ohh, yes. Thanks, code reuse makes sense. I have added changes that leverage this.